### PR TITLE
Closes #932 set new account creation to admin only

### LINKED ISSUE #932: Anonymous visitors can create accounts (w/ admin approval)
## Problem/Motivation
At the moment, visitors can create accounts on a new build (but admin approval is required to activate the account). The settings can be found here: /admin/config/people/accounts under "Registration and cancellation"

## Proposed resolution
By default, have new builds have the "Who can register accounts?" set to "Administrators only".

### PR DESCRIPTION
**Pull request by trackleft**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: az_quickstart.profile [hunk 1]
```diff
@@ -4,10 +4,23 @@
...
+/**
+ * Force import of user.settings config.
+ */
+function az_quickstart_update_9202() {
+  // Set account creation rules, default to only admin can create accounts.
+  $config = \Drupal::service('config.factory')->getEditable('user.settings');
+  $config->setData([
+    'register' => 'admin_only'
+  ])
+    ->save();
```
Comment (joeparsons) [inline az_quickstart.profile:24]: Nitpick for readability.  This isn't technically more correct (might actually be less correct coding-standards wise).

```suggestion
  $config->setData(['register' => 'admin_only'])
    ->save();
```

**Comment by trackleft** [comment]
When I run the update, the setting changes here: /admin/config/people/accounts

But I don't see the update reflected here in the database:

```	user.settings	a:9:{s:9:"anonymous";s:9:"Anonymous";s:11:"verify_mail";b:1;s:6:"notify";a:8:{s:14:"cancel_confirm";b:1;s:14:"password_reset";b:1;s:16:"status_activated";b:1;s:14:"status_blocked";b:0;s:15:"status_canceled";b:0;s:22:"register_admin_created";b:1;s:29:"register_no_approval_required";b:1;s:25:"register_pending_approval";b:1;}s:8:"register";s:23:"visitors_admin_approval";s:13:"cancel_method";s:17:"user_cancel_block";s:22:"password_reset_timeout";i:86400;s:17:"password_strength";b:1;s:8:"langcode";s:2:"en";s:5:"_core";a:1:{s:19:"default_config_hash";s:43:"xxxxxxxx";}} ```

**Comment by trackleft** [comment]
```"register";s:23:"visitors_admin_approval"```

**Comment by trackleft** [comment]
Eww this is what happens when I re-export the config 
image

File: az_quickstart.profile [hunk 2]
```diff
@@ -11,3 +11,12 @@
...
   \Drupal::service('module_installer')->install(['az_metrics']);
 }
+
+/**
+ * Set account creation rules, default to admin_only.
+ */
+function az_quickstart_update_9202() {
+  $config = \Drupal::service('config.factory')->getEditable('user.settings');
+  $config->setData(['register' => 'admin_only'])
+    ->save();
```
Comment (joeparsons) [inline az_quickstart.profile:21]: I think this is what we want:
```suggestion
  $config->set('register', 'admin_only')
    ->save(TRUE);
```

Comment (trackleft) [reply az_quickstart.profile]: Tested, now it works correctly RTBC

**Review by joeparsons — APPROVED** [review]
This looks good to me now.  Tested the DB update locally in lando.

**Review by smayden — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**